apps/dotcms-ui/src/app/view/components/dot-content-compare/components/dot-content-compare-table/dot-content-compare-table.component.html

```diff
@@ -0,0 +1,84 @@
+<p-table [value]='data.fields' *ngIf='data' autoLayout="true" styleClass="p-datatable-gridlines">
+    <ng-template pTemplate='header'>
+        <tr>
+            <th> </th>
+            <th>{{data.working?.title ? data.working?.title : data.working?.identifier}}</th>
+            <th class="dot-content-compare-table__controls">
+                <p-dropdown [options]="data.versions" (onChange)='version.emit($event.value)' appendTo="body" >
+                    <ng-template let-selected pTemplate="selectedItem">
+                        {{selected.modDate | date:'MM/dd/YYYY - hh:mm'}}
+                    </ng-template>
+                    <ng-template let-item pTemplate="item">
+                        {{item.modDate | date:'MM/dd/YYYY - hh:mm'}}
+                    </ng-template>
+                </p-dropdown>
+                <p-selectButton
+                    [options]="displayOptions"
+                    [(ngModel)]="showDiff"
+                    class="p-button-tabbed"
+                    (onChange)="diff.emit($event.value)" >
+                </p-selectButton>
+            </th>
+        </tr>
+    </ng-template>
+
+    <ng-template pTemplate="body" let-field >
+        <tr [ngSwitch]="field?.fieldType" >
+
+           <td>{{field.name}}</td>
+            <ng-container *ngSwitchCase="'Image'">
+               <td><img [src]="'/dA/'+data.working[field.variable]+'/250w/20q'"></td>
+               <td><img [src]="'/dA/'+data.compare[field.variable]+'/250w/20q'"></td>
+            </ng-container>
+            <ng-container *ngSwitchCase="'File'">
+                <td>
+                    <dot-content-compare-file-field
+                        [fileId]="data.working[field.variable]"
+                        [label]=" data.compare[field.variable]" >
+                    </dot-content-compare-file-field>
+                </td>
+                <td>
+                    <dot-content-compare-file-field
+                        [fileId]="data.compare[field.variable]"
+                        [label]="showDiff ? (data.working[field.variable] | dotDiff:data.compare[field.variable]) : data.compare[field.variable]" >
```

Comment:
If you already have the `showDiff` in the state just transform the data (diff html etc...) there and that way you avoid having all this complexity in the template.

Having many conditions in the template should be avoided as much as possible, specially if you have a service that can do all that.

Or if you want to keep the pipe which is also cool, pass the `showDiff` to the pipe as well and let it handle both cases.